packages/shared/lib/services/providers.ts

```diff
@@ -42,22 +42,29 @@ export async function monitorProviders(): Promise<() => void> {
     providers = JSON.parse(providersRaw) as Record<string, Provider>;
     logger.info(`Providers loaded from url ${providersUrl} (${providersHash})`);
 
-    const timeout = setInterval(async () => {
-        try {
-            const providersRaw = await fetchProvidersRaw(providersUrl);
-            const newProvidersHash = createHash('sha1').update(providersRaw).digest('hex');
-
-            if (newProvidersHash !== providersHash) {
-                providersHash = newProvidersHash;
-                providers = JSON.parse(providersRaw) as Record<string, Provider>;
-                logger.info(`Providers reloaded (${providersHash})`);
+    let running = true;
+    void (async function () {
```

Comment:
I would be tempted to name the function instead of anonymous call to be a bit more descriptive
```
const pollProviders = async ...
pollProviders()
```
I don't feel strongly about it (was not named before) so up to you

Reply:
I waffled on that - I'll go ahead and do it

Reply:
Fixed in 9eeef416